### PR TITLE
Add xd002 support

### LINKED ISSUE #2734: [discussion] XD-002, a cool, new, but attiny85 2% keyboard
I know what you are all thinking about right now while reading this:

"Oh no, not the tiny !" 
"we can't do anything on that uc !"
"so few space"

But I did a bet with a friend that qmk can run it !

Obvisouly I failed, i wans't able to compile either the timer.c and bootloader.c files with the V-USB stack at the current state of the QMK repository.

But TMK seems to compile actually for attin85, without a lot of functions of course.

So, is anyone able to guide me with re-enabling QMK on attiny85 so I may try to add this nice good looking 2 key keyboard to the repository ?

Many thanks for reading and advices !

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Now with all the prerequisites merged, we can finally add some form of support.

### Future TODOs
- [ ] core bootloader support?
- [ ] save more space (eg bootloader code has "unused" func with eats 90 bytes) 

### Notes
With bootloader, 6.12k firmware space

Bare minimum
```console
Checking file size of xd002_default.hex                                                             [OK]
 * The firmware size is fine - 4014/6330 (63%, 2316 bytes free)
```

Not enough space for...lots of stuff
* rgblight `RGBLIGHT_ANIMATIONS`

As a general rule, you can enable _**one**_ of the following
* extra
* mouse
* tap/hold
* tap dance
* basic rgblight (a single animation *might* fit)
* layers

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #2734

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd002/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# xd002
+
+![xd002](https://cdn.shopify.com/s/files/1/2711/4238/products/xd002_1024x1024.jpg?v=1522134411)
+
+2% Custom mechanical keyboard. ATtiny85 powered, with 2xws2812 LEDs, and the micronucleus bootloader.
```
Comment (drashna) [inline keyboards/xd002/README.md]: Just a readability fix:
```suggestion
2% Custom mechanical keyboard. ATtiny85 powered, with 2xWS2812 LEDs, and the micronucleus bootloader.
```

File: keyboards/xd002/keymaps/rgb/keymap.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  [0] = LAYOUT(
+    RGB_HUI,  QMKURL
+  )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch (keycode) {
+      case QMKURL:
+        // Using SEND_STRING here adds 400bytes ...
```
Comment (fauxpark) [inline keyboards/xd002/keymaps/rgb/keymap.c:18]: :(

File: keyboards/xd002/info.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "keyboard_name": "xd002",
+  "url": "",
```
Comment (Erovia) [inline keyboards/xd002/info.json:3]:
```suggestion
  "url": "https://kprepublic.com/products/xd002-xiudi-2-custom-mechanical-keyboard-2-keys-underglow-and-switch-rgb-pcb-programmed-hot-swappable-macro-key-aluminum-case",
```

File: keyboards/xd002/keymaps/rgb/keymap.c (hunk 2 above)
Comment (zvecr) [reply keyboards/xd002/keymaps/rgb/keymap.c:18]: There are a few other places i plan to trim space from, so one day :crossed_fingers:

**Pull request merged**